src/modules/hip/kernel/flip_voxel.hpp

```diff
@@ -0,0 +1,198 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+template <typename T>
+__global__ void flip_ncdhw_tensor(T *srcPtr,
+                                  uint3 srcStridesCDH,
+                                  T *dstPtr,
+                                  uint3 dstStridesCDH,
+                                  int channels,
+                                  uint3 mirrorXYZ,
+                                  RpptROI3DPtr roiGenericPtrSrc,
+                                  int batchIndex)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;        // W - inner most dim vectorized
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;              // H - second to inner
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;              // D - outer most dim
+
+    if ((id_z >= roiGenericPtrSrc->xyzwhdROI.roiDepth) || (id_y >= roiGenericPtrSrc->xyzwhdROI.roiHeight) || (id_x >= roiGenericPtrSrc->xyzwhdROI.roiWidth))
+    {
+        return;
+    }
+
+    int dstIdx = (id_z * dstStridesCDH.y) + (id_y * dstStridesCDH.z) + id_x;
+    int xFactor =  id_x + roiGenericPtrSrc->xyzwhdROI.xyz.x;
+    int yFactor = (id_y + roiGenericPtrSrc->xyzwhdROI.xyz.y) * srcStridesCDH.z;
+    int zFactor = (id_z + roiGenericPtrSrc->xyzwhdROI.xyz.z) * srcStridesCDH.y;
+
+    if (mirrorXYZ.y)
+        yFactor = (roiGenericPtrSrc->xyzwhdROI.roiHeight - 1 - id_y) * srcStridesCDH.z;
```

Comment:
Instead of first calculating yFactor and then overriding it, can we do this?
```
int yFactor  = (mirrorXYZ.y) ? ((roiGenericPtrSrc->xyzwhdROI.roiHeight - 1 - id_y) * srcStridesCDH.z) : ((id_y + roiGenericPtrSrc->xyzwhdROI.xyz.y) * srcStridesCDH.z)?
```